### PR TITLE
Update develop to 1.18 API

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: src/H5Fpublic.h [hunk 1]
```diff
@@ -189,10 +189,11 @@ typedef enum H5F_libver_t {
     H5F_LIBVER_V112     = 3, /**< Use the latest v112 format for storing objects */
     H5F_LIBVER_V114     = 4, /**< Use the latest v114 format for storing objects */
     H5F_LIBVER_V116     = 5, /**< Use the latest v116 format for storing objects */
+    H5F_LIBVER_V118     = 6, /**< Use the latest v116 format for storing objects */
```
Comment (lrknox) [inline src/H5Fpublic.h]: Should be < Use the latest v118 format for storing objects */ ?

Comment (byrnHDF) [reply src/H5Fpublic.h]: fixed

File: CMakeLists.txt [hunk 2]
```diff
@@ -856,9 +856,9 @@ if (HDF5_ENABLE_SUBFILING_VFD)
 endif()
 
 
-#option (DEFAULT_API_VERSION "Enable v1.16 API (v16, v18, v110, v112, v114, v116)" "v116")
-set (DEFAULT_API_VERSION "v116" CACHE STRING "Enable v1.16 API (v16, v18, v110, v112, v114, v116)")
-set_property (CACHE DEFAULT_API_VERSION PROPERTY STRINGS v16 v18 v110 v112 v114 v116)
+#option (DEFAULT_API_VERSION "Enable v1.16 API (v16, v18, v110, v112, v114, v116, v118)" "v116")
+set (DEFAULT_API_VERSION "v116" CACHE STRING "Enable v1.16 API (v16, v18, v110, v112, v114, v116, v118)")
```
Comment (jhendersonHDF) [inline CMakeLists.txt]: Should update `DEFAULT_API_VERSION` to v118 and probably removed the commented out line above

File: fortran/src/H5_ff.F90
Comment (jhendersonHDF) [inline fortran/src/H5_ff.F90:1]: The size of H5F_flags needs to be bumped on ~line 59 from 31 to 32

File: CMakeLists.txt (hunk 2 above)
Comment (byrnHDF) [reply CMakeLists.txt]: Agree

File: tools/src/h5repack/h5repack.h [hunk 3]
```diff
@@ -119,7 +119,8 @@
  * \li <strong>3</strong> This is #H5F_LIBVER_V112 in #H5F_libver_t struct
  * \li <strong>4</strong> This is #H5F_LIBVER_V114 in #H5F_libver_t struct
  * \li <strong>5</strong> This is #H5F_LIBVER_V116 in #H5F_libver_t struct
- * \li #H5F_LIBVER_LATEST is aliased to #H5F_LIBVER_V116 for this release
+ * \li <strong>5</strong> This is #H5F_LIBVER_V118 in #H5F_libver_t struct
```
Comment (jhendersonHDF) [inline tools/src/h5repack/h5repack.h]: 5 -> 6

File: .github/workflows/main-cmake-spc.yml [hunk 4]
```diff
@@ -199,6 +199,66 @@ jobs:
         run: ctest . --parallel 2 -C Debug -V
         working-directory: ${{ runner.workspace }}/build
 
+  build_v1_14:
+    name: "gcc DBG v1.12 default API"
```
Comment (jhendersonHDF) [inline .github/workflows/main-cmake-spc.yml]: v1.12 -> v1.14, though it looks like we also should have a v1.16 default API build as well?

File: CMakeLists.txt [hunk 5]
```diff
@@ -856,9 +856,8 @@ if (HDF5_ENABLE_SUBFILING_VFD)
 endif()
 
 
-#option (DEFAULT_API_VERSION "Enable v1.16 API (v16, v18, v110, v112, v114, v116)" "v116")
-set (DEFAULT_API_VERSION "v116" CACHE STRING "Enable v1.16 API (v16, v18, v110, v112, v114, v116)")
-set_property (CACHE DEFAULT_API_VERSION PROPERTY STRINGS v16 v18 v110 v112 v114 v116)
+set (DEFAULT_API_VERSION "v118" CACHE STRING "Enable v1.16 API (v16, v18, v110, v112, v114, v116, v118)")
```
Comment (jhendersonHDF) [inline CMakeLists.txt:859]: v1.16 -> v1.18, though the cache string could probably be revised to not mention the version number

File: HDF5Examples/C/TUTR/testh5cc.sh.in [hunk 6]
```diff
@@ -523,17 +526,32 @@ elif [ -n "$H5_USE_114_API_DEFAULT" ]; then
...
     TOOLTEST -DH5_USE_112_API_DEFAULT $v112main
     TOOLTEST $v114main
+elif [ -n "$H5_USE_116_API_DEFAULT" ]; then
+    echo "Testing HDF5  with 118_API_DEFAULT"
+    TOOLTEST -DH5_USE_16_API_DEFAULT $v16main
+    TOOLTEST -DH5_USE_18_API_DEFAULT $v18main
+    TOOLTEST -DH5_USE_110_API_DEFAULT $v110main
+    TOOLTEST -DH5_USE_112_API_DEFAULT $v112main
+    TOOLTEST -DH5_USE_114_API_DEFAULT $v114main
+    TOOLTEST -DH5_USE_116_API_DEFAULT $v114main
```
Comment (jhendersonHDF) [inline HDF5Examples/C/TUTR/testh5cc.sh.in]: The above sections only run TOOLTEST up to API_VERSION  - 2, so this line looks like it should be removed, though I don't know if that really affects anything

File: HDF5Examples/C/TUTR/testh5cc.sh.in [hunk 7]
```diff
@@ -523,17 +526,32 @@ elif [ -n "$H5_USE_114_API_DEFAULT" ]; then
     TOOLTEST -DH5_USE_110_API_DEFAULT $v110main
     TOOLTEST -DH5_USE_112_API_DEFAULT $v112main
     TOOLTEST $v114main
+elif [ -n "$H5_USE_116_API_DEFAULT" ]; then
+    echo "Testing HDF5  with 118_API_DEFAULT"
```
Comment (jhendersonHDF) [inline HDF5Examples/C/TUTR/testh5cc.sh.in]: 118 -> 116

File: .github/workflows/main-cmake-spc.yml (hunk 4 above)
Comment (byrnHDF) [reply .github/workflows/main-cmake-spc.yml]: Oh I thought updated this, must have forgot to save

**Review by jhendersonHDF — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

**Pull request merged**